### PR TITLE
Da/debug error handling

### PR DESCRIPTION
**Pull request by danamansana**
This fixes a bug with making a request to Sierra to see existing patron holds. The relevant code is line 51 of models/request_result which uses `SierraRequest.build_new_sierra_request({})` instead of `SierraRequest.new({}) `. This ensures that the base url for the SierraRequest will be set. In addition, I have added some more logging and error handling

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,15 +72,15 @@ create_qa:
...
-		
+
 deploy_qa: ## Deploys the latest version to AWS QA
 	@cp config/var_qa.env config/var_deploy.env
 	@export AWS_DEFAULT_PROFILE=qa
 	@make package
 	aws lambda update-function-code \
 		--function-name RecapHoldRequestConsumer-qa \
 		--zip-file fileb://./deploy/rhrc.zip \
-		--profile qa
```
Comment (emu47) [inline Makefile]: Can you create a profile locally instead of changing this? OR... make it consistent with the create script?

File: Makefile [hunk 2]
```diff
@@ -72,15 +72,15 @@ create_qa:
 		--role arn:aws:iam::946183545209:role/lambda-full-access \
 		--zip-file fileb://./deploy/rhrc.zip \
 		--profile qa
```
Comment (emu47) [inline Makefile]: profile qa ... see comment below. this should match.

File: models/request_result.rb [hunk 3]
```diff
@@ -46,12 +46,20 @@ def self.already_sent_error?(message_hash)
...
+      sierra_request.get_holds(patron)
+    rescue Error => e
+      CustomLogger.new("level" => "ERROR", "message" => "Unable to get holds for patron: #{e.message}")
+    end
+  end
+
   def self.patron_already_has_hold?(hold_request)
     patron = hold_request["data"]["patron"]
     record = hold_request["data"]["record"]
-    sierra_request = SierraRequest.new({})
```
Comment (emu47) [inline models/request_result.rb]: wondering if you need to do the whole "build_new_sierra_request" if you're just testing if the patron already has the hold.

Comment (danamansana) [reply models/request_result.rb]: As far as I can tell the bug this branch is responding to was caused by `sierra_request` not having a base url and trying to hit a nonsense endpoint. Just using `.new` doesn't initialize the base url but `build_new_sierra_request` does.

File: Makefile (hunk 2 above)
Comment (danamansana) [reply Makefile]: Good point. See above.

File: Makefile (hunk 1 above)
Comment (danamansana) [reply Makefile]: I think most people have nypl-digital-dev profiles. If it's the same to you I'll change the create script to be consistent

File: models/request_result.rb (hunk 3 above)
Comment (emu47) [reply models/request_result.rb]: Ah. I see. In that case, I'd suggest adding this line right after line 52, then. 

`sierra_request.base_request_url = ENV['SIERRA_URL']`

Reason being I think the .build_new_sierra_request does some logging that I don't think you want right here.

File: Makefile (hunk 1 above)
Comment (emu47) [reply Makefile]: The profile is configured locally (for me is just another entry I created at `~/.aws/credentials`). You can set it up however you like, right? The concern here is that you don't want to confuse qa and production at the point of deployment, so I separated them. nypl-digital-dev is fine. I just prefer keeping them separate for peace of mind.

File: models/request_result.rb (hunk 3 above)
Comment (danamansana) [reply models/request_result.rb]: Changed

**Pull request merged**